mama/c_cpp/src/c/mama/publisher.h

```diff
@@ -196,8 +283,61 @@ extern mama_status
 mamaPublisher_getTransport (mamaPublisher   publisher,
                             mamaTransport*  transport);
 
+/**
```

Comment:
Yeah I think letting the application decide how to allocate the callback struct is better - it's how we deal with mamaMsgCallbacks today as well. Would be a different story if the function which was receiving that pointer didn't effectively memcpy the struct anyway.